### PR TITLE
Dat 715 ingregate send invoice to sap

### PR DESCRIPTION
**Pull request by msimionati**
send the invoice created during the checkout process to sap

### CODE REVIEW

File: Doppler.BillingUser.Test/PostAgreementTest.cs [hunk 1]
```diff
@@ -126,16 +127,28 @@ public async Task POST_agreement_should_return_Ok_when_total_amount_validation_i
...
+
+            var encryptionServiceMock = new Mock<IEncryptionService>();
+            encryptionServiceMock.Setup(x => x.DecryptAES256(It.IsAny<string>())).Returns("12345");
 
             var client = _factory.WithWebHostBuilder(builder =>
             {
                 builder.ConfigureTestServices(services =>
                 {
-                    services.AddSingleton(Mock.Of<IEncryptionService>());
+                    services.AddSingleton(encryptionServiceMock.Object);
```
Comment (nreal) [inline Doppler.BillingUser.Test/PostAgreementTest.cs:146]: I think that this change is not necessary, you can revert it and the test should be run correctly

File: Doppler.BillingUser.Test/PostAgreementTest.cs [hunk 2]
```diff
@@ -126,16 +127,28 @@ public async Task POST_agreement_should_return_Ok_when_total_amount_validation_i
...
+            billingRepositoryMock.Setup(x => x.GetBillingCredit(It.IsAny<int>())).ReturnsAsync(new BillingCredit()
+            {
+                IdBillingCredit = 1,
+                Date = new DateTime(2021, 12, 10)
+            });
+
+            var sapServiceMock = new Mock<ISapService>();
+            sapServiceMock.Setup(x => x.SendBillingToSap(It.IsAny<SapBillingDto>(), It.IsAny<string>()));
+
+            var encryptionServiceMock = new Mock<IEncryptionService>();
```
Comment (nreal) [inline Doppler.BillingUser.Test/PostAgreementTest.cs:139]: I think that this change is not necessary, you can revert it and the test should be run correctly

File: Doppler.BillingUser/Controllers/BillingController.cs [hunk 3]
```diff
@@ -24,6 +28,11 @@ public class BillingController
         private readonly IAccountPlansService _accountPlansService;
         private readonly IValidator<AgreementInformation> _agreementInformationValidator;
         private readonly IPaymentGateway _paymentGateway;
+        private readonly ISapService _sapService;
+        private readonly IEncryptionService _encryptionService;
+        private readonly IOptions<SapSettings> _sapSettings;
+
```
Comment (jhoffman-ms) [inline Doppler.BillingUser/Controllers/BillingController.cs:34]: quizás podes eliminar esta linea en blanco

File: Doppler.BillingUser/Controllers/BillingController.cs [hunk 4]
```diff
@@ -200,25 +218,66 @@ public async Task<IActionResult> CreateAgreement([FromRoute] string accountname,
...
             {
-                var partialBalance = await _userRepository.GetAvailableCredit(user.IdUser);
-                await _billingRepository.CreateMovementCreditAsync(billingCreditId, partialBalance, user, newPlan);
+                await _sapService.SendBillingToSap(
+                    await MapBillingToSapAsync(encryptedCreditCard, currentPlan, newPlan, authorizationNumber, invoiceId, billingCreditId),
+                    accountname);
             }
 
             // TODO: SEND NOTIFICATIONS
-            // TODO: create invoice in SAP
```
Comment (jhoffman-ms) [inline Doppler.BillingUser/Controllers/BillingController.cs:219]: Elimina este ```TODO```

File: Doppler.BillingUser/Controllers/BillingController.cs (hunk 3 above)
Comment (msimionati) [reply Doppler.BillingUser/Controllers/BillingController.cs:33]: el espacio es para separar variables de constantes

File: Doppler.BillingUser/Controllers/BillingController.cs (hunk 4 above)
Comment (msimionati) [reply Doppler.BillingUser/Controllers/BillingController.cs:224]: eliminado

**Comment by jhoffman-ms** [comment]
👍

**Pull request merged**